### PR TITLE
DR2-1568 Move to Scala 3

### PR DESCRIPTION
**Pull request by MancunianSam**
Replace implicits with given and using
Replace ._ with .* for imports
Use enums where sensible

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import sbtrelease.ReleaseStateTransformations._
 import Dependencies._
 
-lazy val scala2Version = "2.13.12"
+lazy val scala3Version = "3.3.1"
```
Comment (techncl) [inline build.sbt]: The latest Scala 3 LTS version is now 3.3.3

File: src/test/scala/uk/gov/nationalarchives/DynamoFormattersTest.scala [hunk 2]
```diff
@@ -258,13 +260,13 @@ class DynamoFormattersTest extends AnyFlatSpec with TableDrivenPropertyChecks wi
 
   forAll(typeTable) { rowType =>
     s"all${rowType}FieldsPopulated" should "contain all of the fields in DynamoTable" in {
-      val generatedTable = rowType match {
-        case ArchiveFolder => generateFolderDynamoTable()
-        case ContentFolder => generateFolderDynamoTable()
-        case Asset         => generateAssetDynamoTable()
-        case File          => generateFileDynamoTable()
+      val generatedTableElementNames = rowType match {
```
Comment (techncl) [inline src/test/scala/uk/gov/nationalarchives/DynamoFormattersTest.scala]: Do this variable need the word "generated"?

**Review by techncl — COMMENTED** [review]
Looks good. Just some minor comments.

File: src/test/scala/uk/gov/nationalarchives/DynamoFormattersTest.scala (hunk 2 above)
Comment (MancunianSam) [reply src/test/scala/uk/gov/nationalarchives/DynamoFormattersTest.scala]: No, I don't think so. I've taken it out.

**Review by techncl — APPROVED** [review]
👍

**Pull request merged**